include/dlaf/common/assert.h

```diff
@@ -0,0 +1,38 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <exception>
+#include <iostream>
+
+#include "dlaf/common/utils.h"
+
+#define DLAF_CHECK(category, condition, ...)                                                           \
```

Comment:
Hmm, can we actually move all the logic into the lowest level assert (e.g. DLAF_PRECONDITION) and scrap `DLAF_CHECK` entirely?